### PR TITLE
ci: add release action on push to master (IN-306)

### PR DESCRIPTION
**Pull request by Xavientois**
**Part of IN-306**

### Brief description. What is this change?

Adds a GitHub Actions Workflow to create a new release when changes are pushed to master. This follows the rules of semantic commits, so not all commits will create a new release.

Also updates the CircleCI config to use the new publish command to publish on release.

### CODE REVIEW

**Review by xavidop — CHANGES_REQUESTED** [review]
I like this! good job!

File: .github/release.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Conventional Changelog Action
+        id: changelog
+        uses: TriPSs/conventional-changelog-action@v3
+        with:
+          output-file: false
+          tag-prefix: voiceflow/common@
```
[the remark this answers is not in the document]
Comment (Xavientois) [reply .github/release.yaml:24]: Good catch, this is not actually the prefix we want. I will remove it in favour of the default `v` that actually matches what is in the CircleCI docs

File: .github/release.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+          skip-commit: true
+          skip-version-file: true
+
+      - name: Create Release
```
[the remark this answers is not in the document]
Comment (Xavientois) [reply .github/release.yaml:28]: The previous job just generates the release description

[the remark this answers is not in the document]
Comment (Xavientois) [reply .github/release.yaml:28]: This pattern is from the examples in the Readme [here](https://github.com/TriPSs/conventional-changelog-action)

**Pull request merged**